src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```
@@ -0,0 +1,127 @@
+     * @param bam file to load
+     * @return RDD of (SAMRecord-backed) GATKReads from the file.
+     */
+    public JavaRDD<GATKRead> getSerialReads(final String bam) {
+        final SAMFileHeader readsHeader = getHeader(bam);
+        List<SimpleInterval> intervals = IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
+        final SamReaderFactory samReaderFactory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT);
+
+        ReadsDataSource bam2 = new ReadsDataSource(new File(bam), samReaderFactory);
+        bam2.setIntervalsForTraversal(intervals);
+        List<GATKRead> records = Lists.newArrayList();
+        for ( GATKRead read : bam2 ) {
+            records.add(read);
+        }
+        return ctx.parallelize(records);
+    }
+
+    /**
+     * Loads Reads using Hadoop-BAM. For local files, bam must have the fully-qualified path,
+     * i.e., file:///path/to/bam.bam.
+     * @param bam file to load
+     * @param intervals intervals of reads to include.
+     * @return RDD of (Google Read-backed) GATKReads from the file.
+     */
+    public JavaRDD<GATKRead> getParallelReads(final String bam, final List<SimpleInterval> intervals) {
+        JavaPairRDD<LongWritable, SAMRecordWritable> rdd2 = ctx.newAPIHadoopFile(
+                bam, AnySAMInputFormat.class, LongWritable.class, SAMRecordWritable.class,
+                new Configuration());
+
+        return rdd2.map(v1 -> {
+            SAMRecord sam = v1._2().get();
+            if (samRecordOverlaps(sam, intervals)) {
+                try {
+                    Read read = ReadUtils.makeRead(sam);
+                    if (read == null) {
+                        throw new GATKException("null read, initial sam: " + sam);
+                    }
+                    return (GATKRead) (GoogleGenomicsReadToGATKReadAdapter.sparkReadAdapter(read));
+                } catch (SAMException e) {
+                    // do nothing if silent
```

Do we really want to silently swallow all errors here? Recommend at least some `logger.error()` output.